src/DomainReader.ts

```diff
@@ -108,6 +108,58 @@ export class DomainReader {
     throw Error(ERROR_MESSAGES.RESOLVER_NOT_SUPPORTED)
   }
 
+  public getAllRolesHistory = async (): Promise<IRoleDefinition[]> => {
+    const network = await this._provider.getNetwork();
+    const chainId = network.chainId;
+    const resolversForChain = this._knownEnsResolvers[chainId];
+    const getEnsAddress = (resolverContractType: ResolverContractType) => {
+      return Object.keys(resolversForChain).find(key => resolversForChain[key] === resolverContractType) || "";
+    }
+
+    const resolvers = [
+      {
+        resolver: PublicResolver__factory.connect(getEnsAddress(ResolverContractType.PublicResolver), this._provider),
+      },
+      {
+        resolver: RoleDefinitionResolver__factory.connect(getEnsAddress(ResolverContractType.RoleDefinitionResolver_v1), this._provider),
+        additionalParser: this.readRoleDefResolver_v1,
+      },
+    ];
+
+    const rolesDefinition = await Promise.all(
+      resolvers.map(async ({ resolver, additionalParser }) => {
+        if (!resolver) return [];
+        const filter = {
+          fromBlock: 0,
+          toBlock: "latest",
+          ...resolver.filters.TextChanged(),
+        };
```

Comment:
I don't think you can rely on just the `TextChanged` event. This is why we have the `DomainNotifier` contract. I think you should just be able to filter for:
1. `TextChanged` events from the `PublicResolver`
2. `DomainUpdated` events from the `DomainNotifier`